# Diffraction par Rayons-X

Lorsque des rayons sont envoyés sur un cristal et si la longueur d'onde est de l'ordre des dimensions de la maille du cristal (typiquement 1A-2A, donc des rayons-X), ceux-ci sont diffractés et subissent des interférences dûes à la structure du cristal. Certaines directions présenteront des interférences constructives dont on pourra mesurer expérimentalement l'intensité de l'onde diffractée. Mesurer dans toutes les directions permet de réaliser un **diffractogramme** dont on va se servir afin d'obtenir toutes les informations de la structure cristalline (motif et réseau). Dans ce notebook sera étudié les trois premiers pics du diffractogramme du cristal CdBr2 pour un rayonnement [$Cu-K\alpha$](http://gisaxs.com/index.php/Cu_K-alpha) correspondant à une **longueur d'onde  de 1.54060Å**.

In [1]:
from pymatgen.ext.matproj import MPRester
from pymatgen.io.cif import CifWriter
from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
from pymatgen.analysis.diffraction.xrd import XRDCalculator
from pymatgen.util.plotting import pretty_plot

# Permet d'utiliser l'interface du MP
with MPRester("2r9TvKUOgDsTx87F")as mpr:
    
    # Permet d'obtenir l'objet 'structure' sur le MP
    structure = mpr.get_structure_by_material_id("mp-568690")
    new_s=SpacegroupAnalyzer(structure).get_primitive_standard_structure()
    
    # Crée un objet permettant de convertir la structure en fichier cif
    cwsym=CifWriter(structure,0.1) # Maille hexagonale
    cwsym.write_file("CdBr2sym")
    cwprim=CifWriter(new_s)     #Maille primitive
    cwprim.write_file("CdBr2prim")
       

## Diffraction dans un cristal
Lorsque les rayons-X intéragissent avec le cristal, l'intensité des rayons diffractés va principalement dépendre de deux paramètres:
* Le réseau cristallin: Les interférences constructives ne vont apparaître que pour certaines directions du rayon incident tandis que pour les autres directions l'amplitude du rayon diffracté sera nulle. Cette condition est géométrique et est formulée mathématiquement par la **loi de Bragg** 

* Le contenu de la maille: Même si les ondes diffusées repectent la loi de Bragg elles interfèrent également avec les atomes présents dans chaque maille induisant une diminution de son intensité. L'intensité résultante sera donc modélisée par le **facteur de structure** de la maille


### Loi de Bragg

Condition nécessaire à l'apparition de pics sur le diffractogramme, elle vient du fait que les rayons incidents sont considérés comme réfléchis par une même famille ${hkl}$ de plans réticulaires. Celà engendre un déphasage entre les ondes réfléchies dont les interférences seront constructives dans certaines conditions donnée par la loi de Bragg:

**Loi de Bragg:** $n\lambda =2d_{hkl}sin\theta$

* $\lambda$: Longueur d'onde du rayon-X
* $\theta$: Angle formé par le plan réticulaire et le rayon incident (ou réfléchi)
* $d_{hkl}$: Distance minimale entre deux plan d'une même famille
* $n$: Nombre entier, appelé aussi "ordre de la réflexion"



### Facteur de structure

Le facteur de structure $S_k$ permet de calculer l'intensité du rayon diffracté (si la loi de Bragg est respectée) et est donné par:

$S_K=\sum_{j=1}^N f_j(K)e^{iK.r'_j}$ où $N$ est le nombre d'atomes dans la maille primitive.

* $K$: Vecteur de diffusion avec $K=k'-k$ où $k$ représente le vecteur d'onde du rayon incident et $k'$ celui du rayon diffusé
* $f_j$: Facteur de forme atomique de l'atome d'indice j de la maille primitive, sa valeur dépend notamment du vecteur de diffusion $K$ et de la densité des charges électroniques autour du noyau de l'atome.
* $r'_j$: Position de l'atome d'indice j dans la maille primitive centrée en (0,0,0)

En conclusion, la position des pics sur un diffractogramme sera déterminée par la condition de Bragg et l'intensité par le facteur de structure tel que $I=|S_K|^2$.

## Analyse des pics du diffractogramme


In [8]:
 #Permet d'obtenir les informations relatives au diffractogramme
xrd=XRDCalculator(1.54060)
xrdpatt=xrd.get_pattern(new_s)

 #1er Pic de diffraction
deth1=xrdpatt.x[0]
I1=xrdpatt.y[0]
Plan1=xrdpatt.hkls[0]
Plan1=Plan1[0]['hkl']
d1=xrdpatt.d_hkls[0]
    
    #2e Pic de diffraction
deth2=xrdpatt.x[1]
I2=xrdpatt.y[1]
Plan2=xrdpatt.hkls[1]
Plan2=Plan2[0]['hkl']
d2=xrdpatt.d_hkls[1]
    
    #3e Pic de diffraction
deth3=xrdpatt.x[2]
I3=xrdpatt.y[2]
Plan3=xrdpatt.hkls[2]
Plan3=Plan3[0]['hkl']
d3=xrdpatt.d_hkls[2]
    
print('Premier pic:')
print('Angle deuxtheta: ',deth1,'°')
print('Intensité: ',I1)
print('Plans réticulaires: ',Plan1)
print('Distance inter-plan: ',d1 )
print()
print('Deuxieme pic:')
print('Angle deuxtheta: ',deth2,'°')
print('Intensité: ',I2)
print('Plans réticulaires: ',Plan2)
print('Distance inter-plan: ',d2 )
print()
print('Troisième pic:')
print('Angle deuxtheta: ',deth3,'°')
print('Intensité: ',I3)
print('Plans réticulaires: ',Plan3)
print('Distance inter-plan: ',d3 )
print()

Premier pic:
Angle deuxtheta:  12.907039872217707 °
Intensité:  63.97073814853898
Plans réticulaires:  (1, 1, 1)
Distance inter-plan:  6.8533759398272105

Deuxieme pic:
Angle deuxtheta:  25.575748472872174 °
Intensité:  24.13492690272719
Plans réticulaires:  (1, 0, 0)
Distance inter-plan:  3.480132967897426

Troisième pic:
Angle deuxtheta:  25.98156774907488 °
Intensité:  1.3074870788910256
Plans réticulaires:  (2, 2, 2)
Distance inter-plan:  3.4266879699136052



Il est également à noter que le [module](https://pymatgen.org/pymatgen.analysis.diffraction.xrd.html) de la librairie pymatgen permettant d'analyser la diffraction rayons-X ajoute la [correction de polarisation de Lorentz](https://dictionary.iucr.org/Lorentz%E2%80%93polarization_correction) pour calculer l'intensité des pics. Les trois premiers pics du diffractogramme sont fournis dans le tableau et graphe suivant:

| $(h k l)$ | $2\theta$ | $I$   | $d_{hkl}$   |
|----------:|-----------|-------|----------------|
| (1 1 1)   | 12.91°    | 63.97 | 6.854          |
| (1 0 0)   | 25.57°    | 24.13 | 3.480          |
| (2 2 2)   | 25.98°    | 1.307 | 3.437          |    

|<img src="Images/xrdplot.PNG" alt="drawing" width="800"/>|
|:---:|
|Intensité relative par rapport à l'onde incidente des différentes valeurs $2\theta<90°$ du diffractogramme|

Où $d_{hkl}$ est exprimé en Angstrom.

Il faut noter que la famille de plans $(1 1 1)$ n'est pas celle de ${2 2 2}$ comme le prouve leurs distances interplan respectives, elles sont en fait intercalées les unes par rapport aux autres comme le montre ce schéma:

|<img src="Images/plans.PNG" alt="drawing" width="500"/>|
|:---:|
|Les plans $(1 1 1)$ sont représentés en bleu et les plans $(2 2 2)$ en rouge et sont formés soit par les atomes Cd (en mauve) ou les atomes Br (en bleu). Dans le cas des plans $(2 2 2)$ les atomes intéragissent avec les électrons de manière plus prononcée entre les plans ce qui dimininue fortement l'intensité de l'onde|